File: app/src/main/java/com/goranch/publicapis/ui/food/details/DetailsFoodModule.java
package com.goranch.publicapis.ui.food.details;

import com.goranch.publicapis.api.FoodService;
import com.goranch.publicapis.api.NutritionService;
import com.goranch.publicapis.di.scopes.FoodDetailScope;
import com.goranch.publicapis.ui.food.repository.FoodDataRepositoryImpl;

import dagger.Module;
import dagger.Provides;

@Module
public class DetailsFoodModule {
    private final DetailRecipeView mDetailRecipeView;

    public DetailsFoodModule(DetailRecipeView detailRecipeView) {
        mDetailRecipeView = detailRecipeView;
    }

    @Provides
    @FoodDetailScope
    DetailRecipeView provideDetailRecipeView() {
        return mDetailRecipeView;
    }

    @Provides
    @FoodDetailScope
    FoodDataRepositoryImpl provideFoodDataRepo(FoodService foodService, NutritionService nutritionService) {
        return new FoodDataRepositoryImpl(foodService, nutritionService);
    }
}

File: app/src/main/java/com/goranch/publicapis/ui/food/details/DetailsFoodModule.kt
package com.goranch.publicapis.ui.food.details

import com.goranch.publicapis.api.FoodService
import com.goranch.publicapis.api.NutritionService
import com.goranch.publicapis.di.scopes.FoodDetailScope
import com.goranch.publicapis.ui.food.repository.FoodDataRepositoryImpl

import dagger.Module
import dagger.Provides

@Module
class DetailsFoodModule(private val mDetailRecipeView: DetailRecipeView) {

    @Provides
    @FoodDetailScope
    internal fun provideDetailRecipeView(): DetailRecipeView {
        return mDetailRecipeView
    }

    @Provides
    @FoodDetailScope
    internal fun provideFoodDataRepo(foodService: FoodService, nutritionService: NutritionService): FoodDataRepositoryImpl {
        return FoodDataRepositoryImpl(foodService, nutritionService)
    }
}
